src/MaxText/model_creation_utils.py

```diff
@@ -112,17 +117,23 @@ def create_model(config, mesh, model_mode: str = MODEL_MODE_TRAIN, rngs: nnx.Rng
   return model
 
 
-def create_nnx_model(config, devices=None):
+def create_nnx_model(config, mesh=None, devices=None, model_mode=None, rng_key=None):
   """Creates a NNX model with sharded parameters, possibly loading from a checkpoint."""
 
-  def _create_model():
-    init_rng = jax.random.PRNGKey(config.init_weights_seed)
-    return from_config(config, devices, rngs=nnx.Rngs(params=init_rng, dropout=1))
+  def _create_model(mesh: Mesh | None = None, model_mode: str | None = None, rng_key: jax.Array | None = None):
+    if rng_key is None:
+      rng_key = jax.random.PRNGKey(config.init_weights_seed)
+    rngs = nnx.Rngs(params=rng_key, dropout=1)
```

Comment:
i don't understand this interface so i'll just flag that this unconditional `dropout=1` scares me a little

Reply:
I needed to remove the explicit `dropout=None` since this was causing errors.